ddtrace/settings/_inferred_base_service.py

```diff
@@ -0,0 +1,203 @@
+import fnmatch
+import os
+import pathlib
+import re
+import sys
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+INIT_PY = "__init__.py"
+ALL_PY_FILES = "*.py"
+
+CACHE: Dict[str, Optional[str]] = {}
+
+
+class DetectionContext:
+    def __init__(self, envs):
+        self.envs = envs
+
+
+class ServiceMetadata:
```

Comment:
same here, do you see more service metadata being added in the future?

Could this just be a NamedTuple instead?

Reply:
Possibly??

Reply:
Im going to leave this, doesn't hurt

Reply:
yeah, I think because of the volume of creation of `ServiceMetadata` is low it isn't a problem.

The reason to consider a `NamedTuple` or alternative is they are lighter weight memory wise, so if you needed to create a ton of these or create them in a tight loop, could be better for memory/GC pressure.